p2p/peering.go

```diff
@@ -497,6 +511,52 @@ func (mw *msgWriter) write(ctx context.Context, msg wire.Message, pver uint32) e
 	}
 }
 
+func (rp *RemotePeer) sendInventory(ctx context.Context) error {
+	timeout := func() time.Duration {
+		const d = maxInvTrickleTimeout - minInvTrickleTimeout
+		return minInvTrickleTimeout + rand.Duration(d)
+	}
+	timer := time.NewTimer(timeout())
+
+	for {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		case <-timer.C:
+			timer.Reset(timeout())
+			if err := rp.sendWaitingInventory(ctx); err != nil {
+				return err
+			}
+		}
+	}
+}
+
+func (rp *RemotePeer) sendWaitingInventory(ctx context.Context) error {
+	rp.waitingInvsMu.Lock()
+	defer rp.waitingInvsMu.Unlock()
+
+	sendInvMsg := func(invList []*wire.InvVect) error {
+		msg := new(wire.MsgInv)
+		msg.InvList = invList
+		return rp.sendMessageAck(ctx, msg)
+	}
+
+	s := rp.waitingInvs
+	rand.ShuffleSlice(s)
```

Comment:
As mentioned, this opens up easy ways for malicious peers to hinder propagation.  Please don't intentionally send invs out of dependency order.

Reply:
Sorry bout that